text/0567-use-and-resources.md

```diff
@@ -0,0 +1,1014 @@
+`ember-concurrency-decorators` and `@ember/render-modifiers`, but it is a bit
+more convoluted:
+
+```js
+export default class SearchResults extends Component {
+  @task
+  *products(url) {
+    let response = yield fetch(url);
+    let data = yield response.json();
+    return data;
+  }
+}
+```
+
+```hbs
+<div
+  {{did-insert (perform this.products @searchUrl)}}
+  {{did-update (perform this.products @searchUrl)}}
+>
+</div>
+```
+
+We can see that this has a few disadvantages compared to the original approach:
+
+- It requires us to switch contexts between the template and the JS to
+  understand the flow of the component.
+- It requires us to explicitly list all of the dependencies for a task _twice_.
+  This is reminiscent of dependency lists in computed properties and observers,
+  and has similar drawbacks in some cases.
+- It requires us to add an element to our template, one which potentially does
+  nothing at all other than exist to allow us to invoke the modifiers.
+
+This approach does have one advantage over the original - it will only call
+perform if `@searchUrl` has actually changed. This would have potentially been a
+subtle bug in the original, since it could cause search results to reload if
+unrelated args changed. But the extra weight and other caveats make this
+solution _feel_ worse overall, as we've discovered through the experience of
+Octane's early adopters.
+
+### Introducing `@use` and Resources
+
+There are two core issues here:
+
+1. With Glimmer components, there is no simple way to define behavior that has
+   its own self contained lifecycle, independent of a template.
+2. Historically, even before Glimmer components, there has been no standard way
+   to mix imperative and declarative code, and expose the results naturally.
+   `ember-concurrency` comes close, but only solves one specific problem.
+
+The `@use` API aims to solve both of these issues, and to do so in a way that
+integrates with autotracking so that users don't have to list out dependencies
+constantly in a repetitive manner. The core of the API consists of:
+
+- The `@use` decorator (and the complementary `{{use}}` helper)
+- The `Resource` base class (implemented via a `ResourceManager`, similar to
+  both components and modifiers).
+
+With these, we can create the `RemoteData` example at the beginning of this RFC
+like so:
+
+```js
+class RemoteData {
+  @tracked data = null;
+  @tracked isLoading = false;
+
+  get state() {
+    return {
+      isLoading: this.isLoading,
+      data: this.data,
+    };
+  }
+
+  async start(url) {
+    this.isLoading = true;
+
+    let response = await fetch(url);
+    let data = await response.json();
+
+    if (!this.isDestroyed) {
+      this.isLoading = false;
+      this.data = data;
+    }
+  }
+}
+
+export const remoteData = resource(RemoteData);
+```
+
+And we would use it like so:
+
+```js
+export default class SearchResults extends Component {
+  @use products = remoteData(this.args.searchUrl);
```

Comment:
Going along with my comment below, I think could be even more clear by renaming it to:

```suggestion
  @use products = remoteDataResource(this.args.searchUrl);
```